x-pack/plugins/reporting/server/core.ts

```diff
@@ -195,21 +195,21 @@ export class ReportingCore {
     return scopedUiSettingsService;
   }
 
-  public getSpaceId(request: KibanaRequest): string | undefined {
+  public getSpaceId(request: KibanaRequest, logger = this.logger): string | undefined {
     const spacesService = this.getPluginSetupDeps().spaces?.spacesService;
     if (spacesService) {
       const spaceId = spacesService?.getSpaceId(request);
 
       if (spaceId !== DEFAULT_SPACE_ID) {
-        this.logger.info(`Request uses Space ID: ` + spaceId);
+        logger.info(`Request uses Space ID: ` + spaceId);
         return spaceId;
       } else {
-        this.logger.info(`Request uses default Space`);
+        logger.debug(`Request uses default Space`);
       }
     }
   }
 
-  public getFakeRequest(baseRequest: object, spaceId?: string) {
+  public getFakeRequest(baseRequest: object, spaceId: string | undefined, logger = this.logger) {
```

Comment:
Even though a logger is always passed in, I wanted to allow `this.logger` to be the default since this will now be the only reference to the logger member of the ReportingCore class. If there were no longer references to it, then it should not be declared. However - that process would create churn since we would want to add the logger back in as a member of the class when we need it while registering the task types, in an upcoming PR.

So, it might not seem important to keep the private `logger` class member, but it helps create a churn of subtractions and additions between different PRs in the code